smdebug/tensorflow/tensor_ref.py

```diff
@@ -96,13 +97,26 @@ def from_variable(cls, variable, export_name=None, mode=None, original_tensor=No
                 # for mirrored variable value this will be the mirrored variable
                 original_tensor = variable
 
-            return TensorRef(
-                variable.value(),
-                export_name=export_name,
-                type=TensorType.VARIABLE,
-                original_tensor=original_tensor,
-                mode=mode,
-            )
+            if version.parse(tf.__version__) >= version.parse("2.0.0") and isinstance(
+                variable, tf.Variable
+            ):
+                # In TF 2.X eager mode, TF throws an error if you try to access a tensor's name.
+                # We need to pass it in as a variable, not a tensor, to maintain the name.
```

Comment:
Why pass it as a variable if it's not a variable? Can you give it  a name instead?

Reply:
in def __init__(), we could try to get the name from tf_obj, else just set the export_name as tensor_ref's name